code/game/machinery/dialysis.dm

```diff
@@ -0,0 +1,91 @@
+/obj/machinery/dialysis
+	name = "dialysis machine"
+	desc = "A machine used to purge reagents from the blood. Simply connect a person to it, turn it on, and wait for the ping."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "rand_machine_1"
+	idle_power_usage = 150
+	active_power_usage = 450
+	anchored = 1
+	density = 1
+	var/list/connections = list()
+	var/max_connections = 1
+	var/reagent_removal_rate = 5
+
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE
+
+/obj/machinery/dialysis/New()
+	. = ..()
+
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/dialysis,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/micro_laser,
+		/obj/item/weapon/stock_parts/micro_laser,
+		/obj/item/weapon/stock_parts/console_screen
+	)
+
+	RefreshParts()
+
+/obj/machinery/dialysis/RefreshParts()
+	var/lasercount = 1
+	var/manipcount = 1
+	for(var/obj/item/weapon/stock_parts/SP in component_parts)
+		if(istype(SP, /obj/item/weapon/stock_parts/micro_laser))
+			lasercount += SP.rating-1
+		if(istype(SP, /obj/item/weapon/stock_parts/manipulator))
+			manipcount += SP.rating-1
+	max_connections = initial(max_connections) * manipcount
+	reagent_removal_rate = initial(reagent_removal_rate) * lasercount
+
+/obj/machinery/dialysis/Destroy()
+	connections.Cut()
+	..()
+
+/obj/machinery/dialysis/MouseDropFrom(over_object, src_location, over_location)
+	if(isobserver(usr))
```

Comment:
observers are already stopped by the incapacitated check so this is not necessary
you might want to allow adminghosts to fiddle with it, instead. perhaps with a warning message to prevent accidents.